### PR TITLE
Add units resources data loading. (Unit test is WIP.)

### PR DESCRIPTION
**Pull request by hugovdm**
- The testGetUnitsData "unit test" is printing for visual inspection, not actually testing yet.
- Having moved this into unitsrouter, it isn't linking yet: "undefined reference".

### CODE REVIEW

**Comment by hugovdm** [comment]
Not really ready for review, it needs to be cleaned up, but I think we decided to merge anyway and then continue developing, more thorough review later?

I'll fix the conflicting files, brb.

File: icu4c/source/i18n/unitsrouter.h [hunk 1]
```diff
@@ -36,6 +37,35 @@ class U_I18N_API UnitsRouter {
...
+
+struct ConversionRateInfo {
+    CharString source;
+    CharString target;
+    CharString factor;
+    CharString offset;
+};
+
+struct UnitPreference {
+    UnitPreference() : geq(0) {}
```
Comment (younies) [inline icu4c/source/i18n/unitsrouter.h]: The default of `geq` should be `1`

**Review by younies — CHANGES_REQUESTED** [review]

File: icu4c/source/i18n/unitsrouter.h (hunk 1 above)
Comment (hugovdm) [reply icu4c/source/i18n/unitsrouter.h]: Strange: github is still telling me "younies requested changes" even though I marked this as "resolved".

Comment (hugovdm) [reply icu4c/source/i18n/unitsrouter.h]: I suspect it's up to you to decide whether your requested changes have been suitably made, it's not up to me.

**Pull request merged**